nemo/collections/tts/models/fastpitch.py

```diff
@@ -404,13 +438,18 @@ def training_step(self, batch, batch_idx):
         return loss
 
     def validation_step(self, batch, batch_idx):
-        attn_prior, durs, speaker = None, None, None
+        attn_prior, durs, speaker, energy = None, None, None, None
         if self.learn_alignment:
             if self.ds_class_name == "TTSDataset":
-                if SpeakerID in self._train_dl.dataset.sup_data_types_set:
-                    audio, audio_lens, text, text_lens, attn_prior, pitch, _, speaker = batch
-                else:
-                    audio, audio_lens, text, text_lens, attn_prior, pitch, _ = batch
+                batch_dict = process_batch(batch, self._train_dl.dataset.sup_data_types_set)
+                audio = batch_dict.get("audio")
+                audio_lens = batch_dict.get("audio_lens")
```

Comment:
Would it be better to represent our batch object as an @dataclass (which is already used a lot in NeMo) instead of a dict?

It is fine if we leave the decision on how to pass these long argument lists around for a later PR.

Reply:
This is a great idea if you mean the `@dataclass` decor from `from dataclasses import dataclass`. Could you please create a ticket and we will address it later?

Reply:
I created a ticket already. So we can track there.